cirq/experiments/fidelity_estimation.py

```diff
@@ -496,32 +556,97 @@ def sample_2q_xeb_circuits(
             is given by this number.
         progress_bar: A progress context manager following the `tqdm` API or `None` to not report
             progress.
+        combinations_by_layer: Either `None` or the result of
+            `rqcg.get_random_combinations_for_device`. If this is `None`, the circuits specified
+            by `circuits` will be sampled verbatim, resulting in isolated XEB characterization.
+            Otherwise, this contains all the random combinations and metadata required to combine
+            the circuits in `circuits` into wide, parallel-XEB-style circuits for execution.
 
     Returns:
-        A pandas dataframe with index given by ['circuit_i', 'cycle_depth'] and
-        column "sampled_probs".
+        A pandas dataframe with index given by ['circuit_i', 'cycle_depth'].
+        Columns always include "sampled_probs". If `combinations_by_layer` is
+        not `None` and you are doing parallel XEB, additional metadata columns
+        will be attached to the returned DataFrame.
     """
```

Comment:
This method became pretty long, it would be good to refactor it and split to smaller sections

Reply:
yes, it's huge. Not helped by the `black` formatter putting everything on its own line! 

Factored out
 - `_get_combinations_by_layer_for_isolated_xeb`
 - `_zip_circuits`
 - `_generate_sample_2q_xeb_tasks`
 - `_execute_sample_2q_xeb_tasks_in_batches`